pkg/controller/core/leader_aware_reconciler.go

```diff
@@ -50,32 +44,27 @@ const defaultRequeueDuration = 15 * time.Second
 //     the leader election lease, in case the previously leading replica failed to renew it.
 func WithLeadingManager(mgr ctrl.Manager, reconciler reconcile.Reconciler, obj client.Object, cfg *config.Configuration) reconcile.Reconciler {
 	// Do not decorate the reconciler if leader election is disabled
-	if cfg.LeaderElection == nil || !ptr.Deref(cfg.LeaderElection.LeaderElect, false) {
+	if !ptr.Deref(cfg.LeaderElection.LeaderElect, false) {
```

Comment:
```suggestion
	if cfg.LeaderElection == nil || !ptr.Deref(cfg.LeaderElection.LeaderElect, false) {
```

I still think that the `WithLeadingManager` should have this verification since this function is exported.

Reply:
Thanks, I've added it back.